mitodl/ocw-studio
# Use Poetry for managing Python Dependencies

### What are the relevant tickets?
#1887 

### Description (What does it do?)
Switches to using `poetry` for Python dependency management.

### How can this be tested?
- Tests should pass on CI
- Everything should work locally after running `docker compose up --build`.
- Testing Heroku deploy will need to wait for RC.
- Try adding a package. You'll need to add/remove the package with poetry CLI, which will make the appropriate edits to  
 `pyproject.toml` and `poetry.lock`:
    - if you have poetry installed on your host: `poetry add/remove --lock <package-name>`. *If you omit `--lock`, the package will be installed on your host, too.*
    - or `docker compose exec -u root web poetry add/remove <package-name>`; (to persist the change between containers, would need to rebuild).
